emcc.py

```diff
@@ -1734,7 +1734,7 @@ def default_setting(name, new_default):
   settings.ASYNCIFY_REMOVE = unmangle_symbols_from_cmdline(settings.ASYNCIFY_REMOVE)
   settings.ASYNCIFY_ONLY = unmangle_symbols_from_cmdline(settings.ASYNCIFY_ONLY)
 
-  if state.mode == Mode.COMPILE_AND_LINK and final_suffix in ('.o', '.bc', '.so', '.dylib') and not settings.SIDE_MODULE:
+  if options.oformat != OFormat.OBJECT and final_suffix in ('.o', '.bc', '.so', '.dylib') and not settings.SIDE_MODULE:
```

Comment:
This looks like more than a deprecation, as it also changes behavior. Or am I confused?

Reply:
This is fixing warning that was showing up erroneously when running with `-r`.     I will split it out.